app/src/main/java/com/amaze/filemanager/filesystem/compressed/CompressedHelper.java

```diff
@@ -117,15 +132,21 @@ public static Decompressor getCompressorInstance(@NonNull Context context, @NonN
     } else if (isTar(type)) {
       decompressor = new TarDecompressor(context);
     } else if (isGzippedTar(type)) {
-      decompressor = new GzipDecompressor(context);
+      decompressor = new TarGzDecompressor(context);
     } else if (isBzippedTar(type)) {
-      decompressor = new Bzip2Decompressor(context);
+      decompressor = new TarBzip2Decompressor(context);
     } else if (isXzippedTar(type)) {
-      decompressor = new XzDecompressor(context);
+      decompressor = new TarXzDecompressor(context);
     } else if (isLzippedTar(type)) {
-      decompressor = new LzmaDecompressor(context);
+      decompressor = new TarLzmaDecompressor(context);
     } else if (is7zip(type)) {
       decompressor = new SevenZipDecompressor(context);
+    } else if (isXz(type)) {
+      decompressor = new CommonCompressedFileDecompressor(context);
+    } else if (isLzma(type)) {
+      decompressor = new CommonCompressedFileDecompressor(context);
+    } else if (isGzip(type)) {
+      decompressor = new CommonCompressedFileDecompressor(context);
```

Comment:
Three branches that do the same thing.

Reply:
If this is feasable please add a comment for why all those files use the file decompressor that provides no info about size and modification.